server/controllers/routes/admin/index.js

```diff
@@ -0,0 +1,22 @@
+const router = require('express').Router();
+const { cohortDelete } = require('./cohort');
+
+router
+  .route('/cohorts/:cohortId')
+  .all((req, res, next) => {
+    // runs for all HTTP verbs first
+    // think of it as route specific middleware!
```

Comment:
@MohammedAlghazali If you know of a future change that you'll put in this `all()` handler (for example, validating that the user has admin privileges) you could put that into a comment, to help others understand why this function exists.
For example, `// TODO: validate the user is an admin (github issue #99)` or something like that.
That would prevent the situation where another coder comes into this file and thinks that `all()` is not doing anything and isn't necessary, and deletes it.

Reply:
thank you very match @tomduggan85 for your notes,
you are right, some member of my team ask me about it  :smile: 
yes, I write it for authentication middleware.